src/object/srv_ec_aggregate.c

```diff
@@ -1468,48 +1358,6 @@ agg_peer_update_ult(void *arg)
 		}
 		ec_agg_in->ea_iod = iod;
 
-		if (entry->ae_cur_stripe.as_ho_ext_cnt ||
-		    !agg_contained(entry)) {
-			ec_agg_in->ea_remove_nr =
-				entry->ae_cur_stripe.as_ho_ext_cnt +
-				entry->ae_cur_stripe.as_extent_cnt;
-			D_ALLOC_ARRAY(ec_agg_in->ea_remove_recxs.ca_arrays,
-				      ec_agg_in->ea_remove_nr);
-			if (ec_agg_in->ea_remove_recxs.ca_arrays == NULL) {
-				rc = -DER_NOMEM;
-				goto out_bulk;
-			}
-			D_ALLOC_ARRAY(ec_agg_in->ea_remove_eps.ca_arrays,
-				      ec_agg_in->ea_remove_nr);
-			if (ec_agg_in->ea_remove_eps.ca_arrays == NULL) {
-				rc = -DER_NOMEM;
-				goto out_bulk;
-			}
-
-			d_list_for_each_entry(ext,
-					      &entry->ae_cur_stripe.as_dextents,
-					      ae_link) {
-				ec_agg_in->ea_remove_recxs.ca_arrays[i] =
-					ext->ae_orig_recx;
-				ec_agg_in->ea_remove_eps.ca_arrays[i++] =
-					ext->ae_epoch;
-			}
-			if (entry->ae_cur_stripe.as_ho_ext_cnt) {
-				d_list_for_each_entry(ext,
-						      &entry->ae_cur_stripe.
-						      as_hoextents,
-						      ae_link) {
-					ec_agg_in->ea_remove_recxs.ca_arrays[i]
-						= ext->ae_orig_recx;
-					ec_agg_in->ea_remove_eps.ca_arrays[i++]
-						= ext->ae_epoch;
-				}
-			}
-			D_ASSERT(ec_agg_in->ea_remove_nr == i);
-			ec_agg_in->ea_remove_recxs.ca_count = i;
-			ec_agg_in->ea_remove_eps.ca_count = i;
-		}
-
```

Comment:
seems you need some other cleanup, such as - remove DAOS_ISEQ_OBJ_EC_AGG's ea_remove_nr/ea_remove_recxs/ea_remove_eps? and such code below:

if (ec_agg_in && ec_agg_in->ea_remove_nr) {
		D_FREE(ec_agg_in->ea_remove_recxs.ca_arrays);
		D_FREE(ec_agg_in->ea_remove_eps.ca_arrays);
	}

Reply:
ah, yes.

Reply:
the RPC input struct, and handler side (ds_obj_ec_agg_handler) also can be cleanup a little bit, but can be done at following patch.